### PR TITLE
Api consumption draft

### PR DESCRIPTION
**Pull request by NSMcEnroe**
## Description
Update: I accidentally pulled your branch into my working branch instead of switching.  That is my bad

This PR does a lot.

-Fixed the structure for facades, services, poros, and controllers.
-Added an argument for https://github.com/facade for search functionality
-Flushed out the movies index
-Installed VCR
-Properly searches for movies
-Added the framework for the show page

Warning: -A lot of new tests need to be done for controller refactoring, sad paths, and edge cases

### Type of Change

- [x] **fix**
- [x] **feat**
- [x] **test**
- [x] **refactor**
- [ ] **docs**

### Checklist

- [x] **Code has been self reviewed**
- [x] **Code runs without any errors**
- [ ] **Thorough testing has been implemented if adding feature**
- [x] **All tests pass**

### CODE REVIEW

File: app/facades/movie_facade.rb [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
+    service = MovieDatabaseService.new
 
-    @movies = data[:results].map do |movie_data|
+    if @keyword.nil?
+      data = service.top_rated_movies
+    else
+      data = service.search_movies(@keyword)
+    end
+
+    data[:results].map do |movie_data|
```
Comment (ttakahashi1591) [inline app/facades/movie_facade.rb:16]: Great logic here to ensure that our routes are directing to the correct API endpoint.

File: app/services/movie_database_service.rb [hunk 2]
```diff
@@ -11,11 +11,10 @@ def get_url(url)
   end
 
   def top_rated_movies
-    get_url("/3/movie/top_rated.json")
+    get_url("/3/movie/top_rated")
   end
 
   def search_movies(keyword)
-    get_url("/3/discover/movie.json")
+    get_url("/3/search/movie?query=#{keyword}")
```
Comment (ttakahashi1591) [inline app/services/movie_database_service.rb:18]: This makes a lot of sense adding the `keyword` query params to the url.
